clang/lib/CIR/CodeGen/CIRGenTypes.cpp

```diff
@@ -168,8 +168,8 @@ mlir::Type CIRGenTypes::convertRecordDeclType(const clang::RecordDecl *RD) {
 
   // Handle forward decl / incomplete types.
   if (!entry) {
-    auto name = getRecordTypeName(RD, "");
```

Comment:
Why `getRecordTypeName()` is getting dropped in favor of `RD->getName()`?

Reply:
I'm actually not sure.

I think it was just because the suffix was not needed so I favored the other for terseness.

I'll switch it back, but considering https://github.com/llvm/clangir/pull/227#discussion_r1294067573, `getRecordTypeName` might be deprecated.

Reply:
Let's keep in mind that we'll likely need to teach lowering about how to produce the same names, for the sake of easy comparison between what differs from traditional LLVM emission. (Btw, Are you tracking any of that as part of this patch set? If not, we probably should (but I'm fine if it comes in later PR)).

For now, can you instead comment out the lines in `getRecordTypeName` that generate the prefixes and add UnimplementedFeatures to track what needs to be added back during lowering?

Reply:
I've removed the prefix addition from `getRecordTypeName` and moved it to the lowering stage to keep it consistent with the original LLVM IR codegen.

Reply:
sounds good